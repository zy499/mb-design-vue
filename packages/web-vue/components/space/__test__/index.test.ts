import { mount } from '@vue/test-utils';
import Space from '../index';

describe('Space', () => {
  test('Size should work', () => {
    const sizes = ['mini', 'small', 'medium', 'large'];
    sizes.forEach((size) => {
      const wrapper = mount(Space, {
        props: { size },
        slots: {
          default: ['<div>aaa</div>', '<div>bbb</div>'],
        },
      });
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
