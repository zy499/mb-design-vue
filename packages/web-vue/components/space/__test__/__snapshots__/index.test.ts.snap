// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space Size should work 1`] = `
"<div class=\\"mb-space mb-space-horizontal mb-space-align-center\\">
  <div class=\\"mb-space-item\\" style=\\"margin-right: 4px;\\">
    <div>aaa</div>
  </div>
  <div class=\\"mb-space-item\\">
    <div>bbb</div>
  </div>
</div>"
`;

exports[`Space Size should work 2`] = `
"<div class=\\"mb-space mb-space-horizontal mb-space-align-center\\">
  <div class=\\"mb-space-item\\" style=\\"margin-right: 8px;\\">
    <div>aaa</div>
  </div>
  <div class=\\"mb-space-item\\">
    <div>bbb</div>
  </div>
</div>"
`;

exports[`Space Size should work 3`] = `
"<div class=\\"mb-space mb-space-horizontal mb-space-align-center\\">
  <div class=\\"mb-space-item\\" style=\\"margin-right: 16px;\\">
    <div>aaa</div>
  </div>
  <div class=\\"mb-space-item\\">
    <div>bbb</div>
  </div>
</div>"
`;

exports[`Space Size should work 4`] = `
"<div class=\\"mb-space mb-space-horizontal mb-space-align-center\\">
  <div class=\\"mb-space-item\\" style=\\"margin-right: 24px;\\">
    <div>aaa</div>
  </div>
  <div class=\\"mb-space-item\\">
    <div>bbb</div>
  </div>
</div>"
`;
